jina/executors/clients.py

```diff
@@ -0,0 +1,145 @@
+             The other methods including ``Classify``, ``Regression``, and ``MultiInference`` are under development.
+
+        """
+        super().__init__(*args, **kwargs)
+        self.model_name = model_name
+        self.signature_name = signature_name
+        self.method_name = method_name
+
+    def post_init(self):
+        """
+        Initialize the channel and stub for the gRPC client
+
+        """
+        self._channel = grpc.insecure_channel('{}:{}'.format(self.host, self.port))
+        from tensorflow_serving.apis import prediction_service_pb2_grpc
+        self._stub = prediction_service_pb2_grpc.PredictionServiceStub(self._channel)
+
+    def get_request(self, data):
+        """
+        Construct the gRPC request to the tf server.
+
+        """
+        request = self.get_default_request()
+        input_dict = self.get_input(data)
+        if self.method_name == 'Predict':
+            return self.fill_request(request, input_dict)
+        else:
+            raise NotImplementedError
+
+    def get_input(self, data) -> Dict:
+        """
+        Convert the input data into a dict with the models input feature names as the keys and the input tensors as the
+            values.
+        """
+        raise NotImplementedError
+
+    def get_response(self, request: 'predict_pb2.PredictRequest'):
+        """
+        Get the response from the tf server and postprocess the response
+        """
+        _response = getattr(self._stub, self.method_name).future(request, self.timeout)
+        if _response.exception():
+            self.logger.error('exception raised in encoding: {}'.format(_response.exception))
+            raise ValueError
+        return self.get_output(_response)
+
+    def get_output(self, response: grpc.UnaryUnaryMultiCallable):
+        """
+        Postprocess the response from the tf server
+        """
+        raise NotImplementedError
+
+    def get_default_request(self) -> 'predict_pb2.PredictRequest':
+        """
+        Construct the default gRPC request to the tf server.
+        """
+        request = self._get_default_request()
+        request.model_spec.name = self.model_name
+        request.model_spec.signature_name = self.signature_name
+        return request
+
+    def _get_default_request(self):
+        if self.method_name == 'Predict':
+            from tensorflow_serving.apis import predict_pb2
+            request = predict_pb2.PredictRequest()
+        elif self.method_name == 'Classify':
+            from tensorflow_serving.apis import classification_pb2
+            request = classification_pb2.ClassificationRequest()
+        elif self.method_name == 'Regression':
+            from tensorflow_serving.apis import regression_pb2
+            request = regression_pb2.RegressionRequest()
+        elif self.method_name == 'MultiInference':
+            from tensorflow_serving.apis import inference_pb2
+            request = inference_pb2.MultiInferenceRequest()
```

Comment:
this part needs to be refactored with `getattr()`, e.g. `getattr(getattr(tensorflow_serving.apis, module_name), request_name)`

Reply:
fixed.